theon/src/main.rs

```diff
@@ -13,21 +13,175 @@
 #![cfg_attr(not(test), no_main)]
 #![cfg_attr(not(test), no_std)]
 
+//!
+//! Theon has two primary responsibilities:
+//!
+//! 1. Taking an inventory of and initializing the hardware so
+//!    that when a description of the machine can be passed when
+//!    control is transferred to the supervisor, and so that the
+//!    system is in a known state.  For example, all processors
+//!    will be parked in the supervisor, running in a
+//!    fully-formed address space, etc.
+//! 2. Loading and initializing all of the consitute binaries
+//!    that make up the system.  That is, the binary images of
+//!    all segments and tasks will be loaded, and segment
+//!    initializers will be invoked.
+//!
+
 extern crate alloc;
 
 mod allocator;
 mod x86_64;
 
+use core::ops::Range;
+
+use arch::{VPageAddr, HPA, MIB, PF4K, V4KA};
+
+type Result<T> = core::result::Result<T, &'static str>;
+
+#[derive(Clone, Copy, Debug)]
+enum BinaryType {
+    Segment,
```

Comment:
Doc comments for these?  What's a Segment vs. Task?

Reply:
Added comments. It's an internal definition, local just to this module (really just for metadata) so not a doc comment.